cargo-quickinstall/src/main.rs

```diff
@@ -34,11 +37,18 @@ fn get_latest_version(crate_name: &str) -> std::io::Result<String> {
             --user-agent 'cargo-quickinstall build pipeline (alsuren@gmail.com)' \
             --location \
             --fail \
-            'https://crates.io/api/v1/crates/{}' \
-            | jq -r .versions[0].num",
+            'https://crates.io/api/v1/crates/{}'",
         crate_name
     );
-    bash_stdout(&command_string)
+    let stdout =  bash_stdout(&command_string)?;
+    let parsed: Result<JsonValue, Error> = match stdout.parse() {
+        Ok(parsed) => Ok(parsed),
+        Err(_) => Err(Error::new(ErrorKind::InvalidData, "Unable to parse JSON."))
+    };
+    let mut version: String = parsed?["versions"][0]["num"].stringify().unwrap();
+    version.remove(0);
+    version.remove(version.len()-1);
```

Comment:
Looks like JsonValue implements `TryInto<String>` https://docs.rs/tinyjson/2.2.0/tinyjson/enum.JsonValue.html#impl-TryInto%3CString%3E . It might be possible to say `let mut version: String = parsed?["versions"][0]["num"].try_into().unwrap()` and avoid the quotes.

```suggestion
    let parsed: Result<JsonValue, Error> = match stdout.parse() {
        Ok(parsed) => Ok(parsed),
        Err(_) => Err(Error::new(ErrorKind::InvalidData, "Unable to parse JSON."))
    };
    let mut version: String = parsed?["versions"][0]["num"].stringify().unwrap();
    version.remove(0);
    version.remove(version.len()-1);
```

L

Reply:
You might also be able to use .map_err() to do the error conversion. Something like:

```suggestion
    let parsed: JsonValue = match stdout.parse().map_err(|_| Err(Error::new(ErrorKind::InvalidData, "Unable to parse JSON.")))?;
    let mut version: String = parsed["versions"][0]["num"].try_into().unwrap();
```

(I've not tested this at all. It probably doesn't even compile 😁 )

My workmate has said that my error reporting "could use some work", so I'm also happy to leave your match statement as it is, and do a proper error-reporting re-work later.